lib/stud/task.rb

```diff
@@ -36,5 +36,14 @@ def stop!
     def self.interrupted?
       Thread.current[:stud_task_interrupted]
     end
+
+    def wakeup
+      @thread.wakeup
```

Comment:
My concern here is that we can wake up not just the loop sleep, but any other sleep the tasks needs to do its job.
Therefore it's necessary to know if the Interval is in the loop sleep stage or not, before calling thread.wakeup